examples/demo-app/src/cloud-providers/dropbox/dropbox-provider.js

```diff
@@ -0,0 +1,475 @@
+// THE SOFTWARE.
+
+// DROPBOX
+import {Dropbox} from 'dropbox';
+import window from 'global/window';
+import Console from 'global/console';
+import DropboxIcon from './dropbox-icon';
+import {MAP_URI} from '../../constants/default-settings';
+import {Provider} from 'kepler.gl/cloud-providers';
+
+const NAME = 'dropbox';
+const DISPLAY_NAME = 'Dropbox';
+const DOMAIN = 'www.dropbox.com';
+const KEPLER_DROPBOX_FOLDER_LINK = `//${DOMAIN}/home/Apps`;
+const CORS_FREE_DOMAIN = 'dl.dropboxusercontent.com';
+const PRIVATE_STORAGE_ENABLED = true;
+const SHARING_ENABLED = true;
+const MAX_THUMBNAIL_BATCH = 25;
+
+function parseQueryString(query) {
+  const searchParams = new URLSearchParams(query);
+  const params = {};
+  for (const p of searchParams) {
+    if (p && p.length === 2 && p[0]) params[p[0]] = p[1];
+  }
+
+  return params;
+}
+
+export default class DropboxProvider extends Provider {
+  constructor(clientId, appName) {
+    super({name: NAME, displayName: DISPLAY_NAME, icon: DropboxIcon});
+    // All cloud-providers providers must implement the following properties
+
+    this.clientId = clientId;
+    this.appName = appName;
+
+    this._folderLink = `${KEPLER_DROPBOX_FOLDER_LINK}/${appName}`;
+    this._path = `/Apps/${window.decodeURIComponent(this.appName)}`;
+
+    // Initialize Dropbox API
+    this._initializeDropbox();
+  }
+
+  /**
+   * This method will handle the oauth flow by performing the following steps:
+   * - Opening a new window
+   * - Subscribe to message channel
+   * - Receive the token when ready
+   * - Close the opened tab
+   */
+  async login(onCloudLoginSuccess) {
+    const link = this._authLink();
+
+    const authWindow = window.open(link, '_blank', 'width=1024,height=716');
+
+    const handleToken = async e => {
+      // TODO: add security step to validate which domain the message is coming from
+      if (authWindow) {
+        authWindow.close();
+      }
+
+      window.removeEventListener('message', handleToken);
+
+      if (!e.data.token) {
+        Console.warn('Failed to login to Dropbox');
+        return;
+      }
+
+      this._dropbox.setAccessToken(e.data.token);
+      // save user name
+      const user = await this._getUser();
+
+      if (window.localStorage) {
+        window.localStorage.setItem(
+          'dropbox',
+          JSON.stringify({
+            // dropbox token doesn't expire unless revoked by the user
+            token: e.data.token,
+            user,
+            timestamp: new Date()
+          })
+        );
+      }
+
+      if (typeof onCloudLoginSuccess === 'function') {
+        onCloudLoginSuccess();
+      }
+    };
+
+    window.addEventListener('message', handleToken);
+  }
+
+  async downloadMap(loadParams) {
+
+    const token = this.getAccessToken();
+    if (!token) {
+      this.login(() => this.downloadMap(loadParams));
+    }
+    const result = await this._dropbox.filesDownload(loadParams);
```

Comment:
try/catch in case of download errors?

Reply:
No need to try catch here Errors will be handled by error action passed to react-palm bimap